packages/trie/test/proof.spec.ts

```diff
@@ -154,3 +155,254 @@ tape('simple merkle proofs generation and verification', function (tester) {
+      useKeyHashing: true,
+    })
+
+    await trie.put(Buffer.from('1000', 'hex'), Buffer.from('a'))
+    await trie.put(Buffer.from('1100', 'hex'), Buffer.from('a'))
+
+    const lKey = Buffer.from('ff'.repeat(32), 'hex')
+    const rKey = Buffer.from('00'.repeat(32), 'hex')
+    try {
+      await trie.createRangeProof(lKey, rKey)
+      t.fail('cannot reach this')
+    } catch (e) {
+      t.pass('succesfully threw')
+    }
+  })
+
+  it('creates one key/value proof', async (t) => {
+    // In this case, there are no key/values between the left and the right key
+    // However, the first value on the right of the rKey key should be reported
+    const trie = new Trie({
+      useKeyHashing: true,
+    })
+
+    const proverTrie = new Trie()
+
+    await trie.put(Buffer.from('1000', 'hex'), Buffer.from('a'))
+    await trie.put(Buffer.from('1100', 'hex'), Buffer.from('a'))
+    await trie.put(Buffer.from('1110', 'hex'), Buffer.from('a'))
+
+    await trie.put(Buffer.from('2000', 'hex'), Buffer.from('b'))
+    await trie.put(Buffer.from('2200', 'hex'), Buffer.from('b'))
+    await trie.put(Buffer.from('2220', 'hex'), Buffer.from('b'))
+
+    await trie.put(Buffer.from('3000', 'hex'), Buffer.from('c'))
+    await trie.put(Buffer.from('3300', 'hex'), Buffer.from('c'))
+    await trie.put(Buffer.from('3330', 'hex'), Buffer.from('c'))
+
+    const lKey = Buffer.from('00'.repeat(32), 'hex')
+    const rKey = Buffer.from('1234' + '00'.repeat(30), 'hex')
+
+    const proof = await trie.createRangeProof(lKey, rKey)
+
+    t.ok(proof.keys.length === 1)
+    t.ok(proof.values.length === 1)
+
+    await proverTrie.verifyRangeProof(
+      trie.root(),
+      proof.keys[proof.keys.length - 1],
+      proof.keys[proof.keys.length - 1],
+      proof.keys,
+      proof.values,
+      proof.proof
+    )
+
+    t.end()
+  })
+
+  it('creates multiple key/value proof', async (t) => {
+    // In this case, report a part of the key/value pairs, together with the double proof of
+    // lKey and rKey
+    const trie = new Trie({
+      useKeyHashing: true,
+    })
+
+    const proverTrie = new Trie()
+
+    await trie.put(Buffer.from('1000', 'hex'), Buffer.from('a'))
+    await trie.put(Buffer.from('1100', 'hex'), Buffer.from('a'))
+    await trie.put(Buffer.from('1110', 'hex'), Buffer.from('a'))
+
+    await trie.put(Buffer.from('2000', 'hex'), Buffer.from('b'))
+    await trie.put(Buffer.from('2200', 'hex'), Buffer.from('b'))
+    await trie.put(Buffer.from('2220', 'hex'), Buffer.from('b'))
+
+    await trie.put(Buffer.from('3000', 'hex'), Buffer.from('c'))
+    await trie.put(Buffer.from('3300', 'hex'), Buffer.from('c'))
+    await trie.put(Buffer.from('3330', 'hex'), Buffer.from('c'))
+
+    const lKey = Buffer.from('00'.repeat(32), 'hex')
+    const rKey = Buffer.from('9999' + '00'.repeat(30), 'hex')
+
+    const proof = await trie.createRangeProof(lKey, rKey)
+
+    // This should report a part of the trie, somewhere between 1 and 9 elements
```

Comment:
Is the return length deterministic? Is it unknown how many key-values are going to be returned or is this just an estimation?

Reply:
It is indeed not known beforehand how much items it will return, I should update this test to better reflect was is going on.